dojo-dungeon/todo-assets/Fehler_Syntax/FehlerhafteKlasse.java

```diff
@@ -1,33 +1,32 @@
-// what is wrong here?
+// import *;
 
-/** This class should be fixed, so that testExcpectedOutput7 and testExcpectedOutput8 work. */
-public class FehlerhafteKlasse { // what is wrong here?
-  int num1 = 5; // what is wrong here?
-  int num2 = 2; // what is wrong here?
+/** This class contains minor syntax errors and should be minimally adjusted to work. */
+public class FehlerhafteKlasse {
+  private final int num1 = 5;
+  private final int num2 = 2;
 
   /**
    * This test should be fixed to print: "Die Summe ist: 7".
    *
    * @param out the output stream to print to.
    */
   public static void redirectOutputTo1(PrintWriter out) {
-    int sum = 0;
-    sum = FehlerhafteKlasse.num1 + FehlerhafteKlasse.num2; // what is wrong here?
-    out.println("Die Summe ist: " + sum);
+    FehlerhafteKlasse fk = null;
+    out.println("Die Summe ist: " + fk.num1 + fk.num2);
   }
 
   /**
    * This test should be fixed to print: "Die dritte Zahl ist: 8".
    *
-   * <p>Tip: Don't forget to use trycatch.
-   *
    * @param out the output stream to print to.
    */
   public static void redirectOutputTo2(PrintWriter out) {
     String perhapsJunk = "10";
-    int num3 = Integer.parseInt(perhapsJunk); // what is wrong here?
-    num3--;
-    num3--;
-    out.println("Die dritte Zahl ist: " + num3);
+    try {
```

Comment:
was genau ist nun die aufgabe hier? soll das fragezeichen im catch ersetzt werden durch 8?

der kommentar obendrüber sagt, ich soll hier "lalala ist 8" ausgeben. dann könnte ich das einfach hart coden und gut ist?

es würde mehr sinn ergeben, wenn die methode einen inputparameter hätte, dessen wert die studis nicht kennen, und den sie nach integer parsen und um 2 dekrementieren sollen und dann mit dem wert eine ausgabe machen sollen. und im fehlerfall soll eine andere ausgabe kommen (die musst du vorgeben). dann kannst du im "compiler" da verschiedene sachen reinstecken ... dann brauchst du in der methode gar nix vorgeben, nur der kommentar muss ordentlich geschrieben werden.

Reply:
> was genau ist nun die aufgabe hier? soll das fragezeichen im catch ersetzt werden durch 8?

Im Prinzip, ja. Mir ist dazu nichts besseres eingefallen.

> 
> der kommentar obendrüber sagt, ich soll hier "lalala ist 8" ausgeben. dann könnte ich das einfach hart coden und gut ist?

Dafür gibt es die Prüfstationen des Dojo-Compilers 1 bis 3. Spätestens bei 3 würde das nicht mehr funktionieren ...

> 
> es würde (wesentlich) mehr sinn ergeben, wenn die methode einen inputparameter hätte, dessen wert die studis nicht kennen, und den sie nach integer parsen und um 2 dekrementieren sollen und dann mit dem wert eine ausgabe machen sollen. und im fehlerfall soll eine andere ausgabe kommen (die musst du vorgeben). dann kannst du im "compiler" da verschiedene sachen reinstecken ... dann brauchst du in der methode gar nix vorgeben, nur der kommentar muss ordentlich geschrieben werden.

Klingt sinnvoll. :)

Reply:
wenn es nicht zuviel arbeit macht, würde ich gern meinen vorschlag mit dem zusätzlichen string-parameter "in auftrag" geben. 

im prinzip brauchst du auch den stream gar nicht, du kannst die methode ja was zurückgeben lassen und abfragen.

Reply:
guter vorschlag, ich hätte eine idee:

```
  public static long calculateFibonacci(String n) {
      // parse n
      // calculate fibonacci
      // return result
  }
```

oder etwas ähnliches

Reply:
ja genau 😀